File: com/problem/cpp/googleKickStart/RecordBreaker2.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
  int t, n;
  cin >> t;

  for (int i = 1; i <= t; i++)
  {
    cin >> n;
    int arr[n], ans = 0, p = 0;

    for (int j = 0; j < n; j++)
    {
      cin >> arr[j];
    }

    for (int j = 0; j < n; j++)
    {
      bool gpd = j == 0 || arr[j] > p;
      bool gnd = j == n - 1 || arr[j] > arr[j + 1];
      
      if (gpd && gnd)
      {
        ans++;
      }
        p = max(p, arr[j]);
    }

    printf("Case #%d: %d\n",i,ans);
  }
}

File: bin/com/problem/python/googleKickstart/2022/session/3/RecordBreaker2.py
t = int(input().strip())
for i in range(1, t+1):
    n = int(input().strip())
    arr = list(map(int, input().split(" ")))
    n = len(arr)
    ans = 0
    p = 0

    for j in range(n):
        gpd = j == 0 or arr[j] > p
        gnd = j == n - 1 or arr[j] > arr[j + 1]
        if (gpd and gnd): ans += 1
        p = max(p, arr[j])

    print("Case #{}: {}".format(i, ans))
